docs/quickstart.fsx

```diff
@@ -0,0 +1,229 @@
+(*** include-output ***)
+
+(**
+ 
+### Data Loaders
+
+A data loader handles tasks such as constructing minibatches from an underlying dataset on-the-fly, shuffling the data, and moving the data tensors between devices. In the example below we show a single batch of six MNIST images and their corresponding classification labels.
+
+*)
+
+let loader = DataLoader(dataset, shuffle=true, batchSize=6)
+let batch, labels = loader.batch()
+
+printfn "%A\nLabels: %A" (batch.toImageString()) labels
+(*** include-output ***)
+
+(**
+
+In practice a data loader would be typically used to iterate over all minibatches in a given dataset in order to feed each minibatch through a machine learning model. One full iteration over the dataset would be called an "epoch". Typically you would perform multiple such epochs of iterations during the training of a model.
+
+*)
+let epochs = 20
+
+for epoch = 1 to epochs do
+    for i, data, labels in loader.epoch() do
+        printfn "Epoch %A, minibatch %A" epoch (i+1)
+        // Process the minibatch
+        // ...
+(**
+
+## Models
+
+Many machine learning models are differentiable functions whose parameters can be tuned via [gradient-based optimization](https://en.wikipedia.org/wiki/Gradient_descent), finding an optimum for an objective function that quantifies the fit of the model to a given set of data. These models are typically built as compositions non-linear functions and ready-to-use building blocks such as linear, recurrent, and convolutional layers.
+
+DiffSharp provides the most commonly used model building blocks including convolutions, transposed convolutions, batch normalization, dropout, recurrent and other architectures.
+
+See the [DiffSharp.Model](/reference/diffsharp-model.html) namespace for the full API reference.
+
+### Constructing models, PyTorch style
+
+If you have experience with [PyTorch](https://pytorch.org/), you would find the following way of model definition familiar. Let's look at an example of a [generative adversarial network (GAN)](https://arxiv.org/abs/1406.2661) architecture.
+*)
+open DiffSharp.Model
+open DiffSharp.Compose
+
+// PyTorch style
+
+// Define a model class inheriting the base
+type Generator(nz:int) =
+    inherit Model()
+    let fc1 = Linear(nz, 256)
+    let fc2 = Linear(256, 512)
+    let fc3 = Linear(512, 1024)
+    let fc4 = Linear(1024, 28*28)
+    do base.add([fc1; fc2; fc3; fc4])
+    override self.forward(x) =
+        x
+        |> dsharp.view([-1;nz])
+        |> fc1.forward
+        |> dsharp.leakyRelu(0.2)
+        |> fc2.forward
+        |> dsharp.leakyRelu(0.2)
+        |> fc3.forward
+        |> dsharp.leakyRelu(0.2)
+        |> fc4.forward
+        |> dsharp.tanh
+
+// Define a model class inheriting the base
+type Discriminator(nz:int) =
+    inherit Model()
+    let fc1 = Linear(28*28, 1024)
+    let fc2 = Linear(1024, 512)
+    let fc3 = Linear(512, 256)
+    let fc4 = Linear(256, 1)
+    do base.add([fc1; fc2; fc3; fc4])
```

Comment:
Maybe put a comment on this line, since it's necessary

Reply:
I agree.

As a separate idea: can we improve the behavior in a way that this `base.add` is not needed? In Python, there is an automated way that the class discovers any attributes that are other NN modules and they automatically are registered as submodules. 

Discussed here:

https://discuss.pytorch.org/t/how-does-a-module-know-about-its-component-modules/13236

and implemented here as a custom `__setattr__`
https://github.com/pytorch/pytorch/blob/394dd391dd5c30de371e77722d5d0261dd90c0b0/torch/nn/modules/module.py#L1138

Can we have a similar thing in F#?

Reply:
Yeah it's hard to remove this without resorting to reflection hackery. I've thought about it a few times and haven't come up with anything